sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -408,6 +418,40 @@ public String getId() {
         return queryItemsInternal(new SqlQuerySpec(query), new CosmosQueryRequestOptions(), classType);
     }
 
+    /**
+     * Initializes the container by warming up the caches and connections.
+     * <br>The execution of this method is expected to result in some RU charges to your account.
+     * The number of RU consumed by this request varies, depending on data consistency, size of the overall data in the container,
+     * item indexing, number of projections. For more information regarding RU considerations please visit
+     * <a href="https://docs.microsoft.com/en-us/azure/cosmos-db/request-units#request-unit-considerations">https://docs.microsoft.com/en-us/azure/cosmos-db/request-units#request-unit-considerations</a>.
+     * @return Mono of Void
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<Void> openConnectionsAndInitCaches() {
+        Mono<List<FeedRange>> feedRangesMono = this.getFeedRanges();
+        AtomicReference<Mono<List<FeedResponse<ObjectNode>>>> sequentialList = new AtomicReference<>();
+        List<Flux<FeedResponse<ObjectNode>>> fluxList = new ArrayList<>();
```

Comment:
Its in lamda that's why used AtomicReference. All flux need to run in parallel for perf and I doesn't want changing.
Please let me know what is harm in this method, also give your exact code, provide perf data base comparison between two approaches. Then i will think over it , otherwise I will keep it as it is

Reply:
> Please let me know what is harm with this approach, also give your exact code, provide perf data base comparison between two approaches. Then i will think over it , otherwise I will keep it as it is

are you asking me to write the exact code and perf test the suggested approach Naveen? :-)

anyway you can take a look at cross-partition query pipeline where we are parallelizing query to multiple partitions. Over there the code has a lot of other knobs as it needs to be more configurable. Your case is simpler.

Reply:
My previous question still stand valid "what is harm with this approach".

Are you refereeing to drainAsync method in query ? 
Based on this i see we are using same list with mergeSequential and passing our custome concurrency https://github.com/Azure/azure-sdk-for-java/blob/663620a678dc125339d7dd44d3fdbcb263ca5544/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java#L388

In this case merge is better the mergeSequential , as we don't need any ordering.
And instead of using custom concurrency i would let reactor to handle 
`	/**
	 * A small default of available slots in a given container, compromise between intensive pipelines, small
	 * subscribers numbers and memory use.
	 */
	public static final int SMALL_BUFFER_SIZE = Math.max(16,
			Integer.parseInt(System.getProperty("reactor.bufferSize.small", "256")));`

Reply:
Naveen, I added this comment when you had AtomicReference and two lists side by side here this commit:
https://github.com/Azure/azure-sdk-for-java/pull/20075/commits/2e75bb277b78f5806cc2d6167fc6cfa98915b974

my comment was more on the AtomicReference and using two lists side by side with custom synchronization logic for ensuring thread safety. I see you addressed part of my comment in a recent commit and removed the custom synchronization AtomicReference:
https://github.com/Azure/azure-sdk-for-java/pull/20075/commits/478c60c3c9398e01c746aacae13b7c416276410a

The intuition behind why I asked to remove AtomicReference, is to rely more on the reactive-stream functionality. If reactive-stream library is providing a functionality we should try to re-use and not re-invent the same concept. This helps in reducing maintenance cost and make code more readable. 

after AtomicReference is removed, there is no problem with using `fluxList` just please move its declaration inside the `flatMap`.

Reply:
Yes I already removed AtomicReference in my previous commit. Also moved fluxList  in flat map in a new Iteration. Thanks